src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -759,7 +759,11 @@ public static float onAnvilRepair(Player player, @NotNull ItemStack output, @Not
     public static int onGrindstoneChange(@NotNull ItemStack top, @NotNull ItemStack bottom, Container outputSlot, int xp)
     {
         GrindstoneEvent.OnplaceItem e = new GrindstoneEvent.OnplaceItem(top, bottom, xp);
-        if (MinecraftForge.EVENT_BUS.post(e)) return e.getXp();
+        if (MinecraftForge.EVENT_BUS.post(e))
+        {
+            outputSlot.setItem(0, ItemStack.EMPTY);
+            return -1;
```

Comment:
Return the `xp` parameter instead (which will be `-1` if fired by vanilla).

Reply:
That would be wrong. XP might have been set before it was cancelled, we want to be certain the result is a known value and not one of our technical values